com.unity.netcode.gameobjects/Components/NetworkTransform.cs

```diff
@@ -713,18 +704,15 @@ public override void OnNetworkSpawn()
             {
                 TryCommitTransformToServer(m_Transform, m_CachedNetworkManager.LocalTime.Time);
             }
+            m_LocalAuthoritativeNetworkState = m_ReplicatedNetworkState.Value;
 
-            // We do this to reset the interpolators so that pooled NetworkObjects will initialize prior
-            // to assigning the current NetworkState -- helps prevent pooled objects from interpolating from the last state
+            // crucial we do this to reset the interpolators so that recycled objects when using a pool will
+            //  not have leftover interpolator state from the previous object
             Initialize();
-
-            m_LocalAuthoritativeNetworkState = m_ReplicatedNetworkState.Value;
```

Comment:
This is most likely the only change needed, no ? The rest occurs on `OnNetworkDespawn` or is just comments.

Reply:
yes but still high risk / undesired — better revert + refix instead of trying to build on top of each other.

Reply:
having said that, "object pooling" is not an SDK feature, this should not have been fixed/addressed in the SDK-level in the first place but here we are :P

Reply:
If this breaks initialization, maybe what's needed is `m_WaitingInit = true` to be set and change the handling when the first value comes in. Although, I'm not familiar enough with the NetworkTransform to tell for sure

Reply:
this works:

```cs
m_LocalAuthoritativeNetworkState = m_ReplicatedNetworkState.Value;

Initialize();
```

this breaks:

```cs
Initialize();

m_LocalAuthoritativeNetworkState = m_ReplicatedNetworkState.Value;
```

but nevertheless, I'm not even interested in fixing this here. we should just revert this out from the release branch, also revert from the develop branch then start over again, then I'm happy to discuss. even then, I wouldn't allow addressing a non-SDK feature with an SDK fix.